diff --git a/src/libANGLE/renderer/wgpu/mac/WindowSurfaceWgpuMetalLayer.mm b/src/libANGLE/renderer/wgpu/mac/WindowSurfaceWgpuMetalLayer.mm
index 548cc2f85..093684e16 100644
--- a/src/libANGLE/renderer/wgpu/mac/WindowSurfaceWgpuMetalLayer.mm
+++ b/src/libANGLE/renderer/wgpu/mac/WindowSurfaceWgpuMetalLayer.mm
@@ -56,7 +56,9 @@
         CGSizeMake(mMetalLayer.bounds.size.width * mMetalLayer.contentsScale,
                    mMetalLayer.bounds.size.height * mMetalLayer.contentsScale);
     mMetalLayer.framebufferOnly  = NO;
+#if ANGLE_PLATFORM_MACOS || ANGLE_PLATFORM_MACCATALYST
     mMetalLayer.autoresizingMask = kCALayerWidthSizable | kCALayerHeightSizable;
+#endif
     mMetalLayer.contentsScale    = layer.contentsScale;
 
     [layer addSublayer:mMetalLayer];
diff --git a/src/libANGLE/renderer/wgpu/wgpu_sources.gni b/src/libANGLE/renderer/wgpu/wgpu_sources.gni
index 726c92169..26fb89c60 100644
--- a/src/libANGLE/renderer/wgpu/wgpu_sources.gni
+++ b/src/libANGLE/renderer/wgpu/wgpu_sources.gni
@@ -68,7 +68,7 @@ if (is_win) {
   ]
 }
 
-if (is_mac) {
+if (is_mac || is_ios) {
   wgpu_backend_sources += [
     "mac/WindowSurfaceWgpuMetalLayer.h",
     "mac/WindowSurfaceWgpuMetalLayer.mm",
